swift/ql/test/query-tests/Security/CWE-022/PathInjectionTest.ql

```diff
@@ -12,10 +12,14 @@ class PathInjectionTest extends InlineExpectationsTest {
   override predicate hasActualResult(Location location, string element, string tag, string value) {
     exists(DataFlow::Node source, DataFlow::Node sink, Expr sinkExpr |
       PathInjectionFlow::flow(source, sink) and
-      sinkExpr = sink.asExpr() and
+      (
+        sinkExpr = sink.asExpr() or
+        sinkExpr = sink.(DataFlow::PostUpdateNode).getPreUpdateNode().asExpr()
+      ) and
```

Comment:
Just seeing this now. Do we even need to talk about the `sinkExpr` here? As far as I can tell, the `sinkExpr` only exists for its `toString` and its location. But we might as well just get that from the `sink` node.

Reply:
`sinkExpr` may correspond to the pre-update node of a post-update `sink`.  We could change `sinkExpr` to be a `DataFlow::Node` but we'd still need the extra variable.

Reply:
What I'm saying is that you could change the body of this predicate to be:
```ql
override predicate hasActualResult(Location location, string element, string tag, string value) {
    exists(DataFlow::Node source, DataFlow::Node sink |
      PathInjectionFlow::flow(source, sink) and
      location = sink.getLocation() and
      element = sink.toString() and
      tag = "hasPathInjection" and
      location.getFile().getName() != "" and
      value = source.asExpr().getLocation().getStartLine().toString()
    )
  }
```
and this would avoid talking about whether the sink is a post-update node or a the pre-update node of a post-update node.

Reply:
Ah I see what you mean now.  All `sinkExpr` really does is force `sink` to be an expr node (or post-update node of an expr node), which we don't particularly want.

I've made your recommended change.  Thanks!